worlds/mmbn3/__init__.py

```diff
@@ -74,10 +79,11 @@ def generate_early(self) -> None:
         if self.options.extra_ranks > 0:
             self.item_frequencies[ItemName.Progressive_Undernet_Rank] = 8 + self.options.extra_ranks
 
+        self.excluded_locations = []
+        if not self.options.include_secret:
+            self.excluded_locations = secret_locations
         if not self.options.include_jobs:
-            self.excluded_locations = always_excluded_locations + [job.name for job in jobs]
-        else:
-            self.excluded_locations = always_excluded_locations
+            self.excluded_locations = self.excluded_locations + [job.name for job in jobs]
```

Comment:
Could do this instead
```suggestion
            self.excluded_locations += [job.name for job in jobs]
```